drivers/gpio/gpio_nrfx.c

```diff
@@ -105,20 +111,40 @@ static int gpiote_pin_int_cfg(const struct device *port, uint32_t pin)
 	/* Pins trigger interrupts only if pin has been configured to do so */
 	if (data->pin_int_en & BIT(pin)) {
 		if (data->trig_edge & BIT(pin)) {
-		/* For edge triggering we use GPIOTE channels. */
-			nrf_gpiote_polarity_t pol;
-
-			if (data->double_edge & BIT(pin)) {
-				pol = NRF_GPIOTE_POLARITY_TOGGLE;
-			} else if ((data->int_active_level & BIT(pin)) != 0U) {
-				pol = NRF_GPIOTE_POLARITY_LOTOHI;
+			if (IS_ENABLED(CONFIG_GPIO_NRF_INT_EDGE_USING_SENSE)) {
+				bool high;
+				uint32_t sense;
+
+				if (nrf_gpio_pin_dir_get(abs_pin) ==
```

Comment:
There's a race condition here if the pin changes state between the fetch and the sense configuration.